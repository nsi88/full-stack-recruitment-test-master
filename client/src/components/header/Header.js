import React from 'react';
import './Header.scss';
import longArrowRight from '../../icons/long-arrow-right.svg';

const Header = ({fromPlace, toPlace, adults, cabinClass}) => (
  <header className='header'>
    <h1 className='header__places'>
      {fromPlace}
      <span className='header__arrow-container'>
        <img className='header__arrow' alt='to' src={longArrowRight}/>
      </span>
      {toPlace}
    </h1>
    <span className='header__details'>{adults}&nbsp;{adults === 1 ? 'adult' : 'travelers'}, {cabinClass}</span>
  </header>
);

export default Header;
