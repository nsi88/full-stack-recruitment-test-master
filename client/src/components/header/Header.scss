@import "~bpk-mixins";

.header {
  display: flex;
  flex-direction: column;
  align-content: center;
  padding: $bpk-spacing-sm;

  background-color: $bpk-color-blue-500;
  color: $bpk-color-white;

  .places {
    @include bpk-heading-1();
    margin-bottom: 0px;
    text-transform: uppercase;

    .arrowContainer {
      padding: $bpk-spacing-sm;

      .arrow {
        width: $bpk-icon-size-sm;
        height: $bpk-icon-size-sm;
        position: relative;
        top: -0.375rem;
      }
    }
  }

  .details {
    @include bpk-text-base();
    text-transform: lowercase;
  }
}
