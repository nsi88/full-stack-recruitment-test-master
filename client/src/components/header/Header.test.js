import React from 'react';
import Header from '../Header';
import renderer from 'react-test-renderer';

test('render', () => {
  const tree = renderer.create(<Header fromPlace="EDI" toPlace="LON" adults="2" cabinClass="Economy"/>).toJSON();
  expect(tree).toMatchSnapshot();
});
