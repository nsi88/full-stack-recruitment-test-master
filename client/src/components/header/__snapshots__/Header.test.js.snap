// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<header
  className="header"
>
  <h1
    className="header__places"
  >
    EDI
    <span
      className="header__arrow-container"
    >
      <img
        alt="to"
        className="header__arrow"
        src="test-file-stub"
      />
    </span>
    LON
  </h1>
  <span
    className="header__details"
  >
    2
    Â 
    travelers
    , 
    Economy
  </span>
</header>
`;
